src/System.Management.Automation/utils/tracing/SysLogProvider.cs

```diff
@@ -0,0 +1,493 @@
+        static string _gitCommitId;
+        // The hash code for the git commit id.
+        static readonly int _commitId;
+
+        [ThreadStatic]
+        static Guid _activity = Guid.NewGuid();
+
+        // Ensure the string pointer is not garbage collected.
+        static IntPtr _nativeSyslogIdent = IntPtr.Zero;
+        static NativeMethods.SysLogPriority _facility = NativeMethods.SysLogPriority.Local0;
+
+        byte _channelFilter;
+        ulong _keywordFilter;
+        byte _levelFilter;
+
+        static SysLogProvider()
+        {
+            // full git commit id string. Will be logged with the associated hash code.
+            _gitCommitId = PSVersionInfo.GitCommitId;
+            // has value for the commit id string.
+            _commitId = _gitCommitId.GetHashCode();
+        }
+
+        public SysLogProvider(string applicationId, PSLevel level, PSKeyword keywords, PSChannel channels)
+        {
+            _nativeSyslogIdent = Marshal.StringToHGlobalAnsi(applicationId);
+            NativeMethods.OpenLog(_nativeSyslogIdent, _facility);
+            _keywordFilter = (ulong)keywords;
+            _levelFilter = (byte) level;
+            _channelFilter = (byte) channels;
+            LogCommit();
+        }
+
+        /// <summary>
+        /// Gets the value indicating if the specified level and keywords are enabled for logging
+        /// </summary>
+        /// <param name="level">The PSLevel to check.</param>
+        /// <param name="keywords">The PSKeyword to check.</param>
+        /// <returns>true if the specified level and keywords are enabled for logging.</returns>
+        internal bool IsEnabled(PSLevel level, PSKeyword keywords)
+        {
+            return
+            (
+                ((ulong) keywords & _keywordFilter) != 0
+                &&
+                ((int) level <= _levelFilter)
+            );
+        }
+
+        // NOTE: There are a number of places where PowerShell code sends analytic events
+        // to the operational channel. This is a side-effect of the custom wrappers that
+        // use flags that are not consistent with the event definition.
+        // To ensure filtering of analytic events is consistent, both keyword and channel
+        // filtering is performed to suppress analytic events.
+        private bool ShouldLog(PSLevel level, PSKeyword keywords, PSChannel channel)
+        {
+            return
+            (
+                (_channelFilter & (ulong)channel) != 0
+                &&
+                IsEnabled(level, keywords)
+            );
+        }
+
+        #region resource manager
+
+        private static global::System.Resources.ResourceManager _resourceManager;
+        private static global::System.Globalization.CultureInfo _resourceCulture;
+
+        private static global::System.Resources.ResourceManager ResourceManager
+        {
+            get
+            {
+                if (object.ReferenceEquals(_resourceManager, null))
```

Comment:
I don't think this is necessary.  You can initialize _resourceManager field where it is defined or in the static constructor.  Also the field can be readonly.

Reply:
This is the pattern used everywhere else.  i believe the notion is to defer loading resources until first use.